community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/AggregationAcceptanceTest.scala

```diff
@@ -243,4 +243,19 @@ class AggregationAcceptanceTest extends ExecutionEngineFunSuite with NewPlannerT
 
     result.toList should equal(List(Map("foo" -> 42, "bar" -> 42, "baz" -> Map("y" -> 1))))
   }
+
+  test("should not project too much when aggregating in a WITH before merge and after a WITH with filter") {
```

Comment:
You could write a query that tries to use a variable that is not in scope, but that, due to the bug, actually _was_ in scope. That query should then fail with some error.